### PR TITLE
[skip ci]Convert Sprint Release into pipeline

### PR DESCRIPTION
**Pull request by PengTian0**
Convert Sprint Relase into pipeline
Depends on: https://github.com/RackHD/on-build-config/pull/103

image

### CODE REVIEW

**Comment by PengTian0** [comment]
@panpan0000 @changev @anhou @yyscamper @keedya

File: jobs/SprintRelease/create_tag.sh [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+--version $new_manifest \
+--file_path $new_manifest
+
+MANIFEST_FILE_URL=https://dl.bintray.com/$BINTRAY_SUBJECT/$BINTRAY_REPO/$new_manifest
+echo "MANIFEST_FILE_URL=<a href=\"$MANIFEST_FILE_URL\">$new_manifest</a>"
+echo "MANIFEST_FILE_URL=$MANIFEST_FILE_URL" > downstream_file
+
+check_file_exist(){
+    echo "try to download the manifest"
+    max_retry=200
```
Comment (panpan0000) [inline jobs/SprintRelease/create_tag.sh:43]: like the retry..

File: jobs/SprintRelease/create_tag.groovy [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+                             passwordVariable: 'BINTRAY_API_KEY', 
+                             usernameVariable: 'BINTRAY_USERNAME')
+        ]){
+
+            if(JUMP_VERSION == "true"){
+                stage("Jump Version"){
+                    sh '''#!/bin/bash -e
+                    export GITHUB_CREDS=$GITHUB_USERNAME:$GITHUB_PASSWORD
+                    
+                    curl --user $BINTRAY_USERNAME:$BINTRAY_API_KEY -L "$MANIFEST_FILE_URL" -o rackhd-manifest
```
Comment (panpan0000) [inline jobs/SprintRelease/create_tag.groovy]: some comments here to explain what does this stage did ?

File: build-release-tools/lib/ManifestGenerator.py [hunk 3]
```diff
@@ -142,3 +142,47 @@ def update_repositories_commit(self, repositories):
...
+        self._force = force
+        self._jobs = jobs
+        self._manifest = Manifest.instance_of_sample()
+        self.repo_operator = RepoOperator(git_credential)
+        self.check_builddir()
+
+    def check_builddir(self):
+        """
+        Checks the given builddir name and force flag.
+        Deletes exists directory if one already exists and --force is set
```
Comment (panpan0000) [inline build-release-tools/lib/ManifestGenerator.py:160]: exists --> existing ?

**Review by panpan0000 — APPROVED** [review]

File: jobs/SprintRelease/create_tag.groovy (hunk 2 above)
Comment (PengTian0) [reply jobs/SprintRelease/create_tag.groovy]: Thanks!
I just added more comments.

File: build-release-tools/lib/ManifestGenerator.py (hunk 3 above)
Comment (PengTian0) [reply build-release-tools/lib/ManifestGenerator.py:160]: Thanks, fixed!

**Review by anhou — APPROVED** [review]

**Pull request merged**